lib/cdo/rake_utils.rb

```diff
@@ -30,6 +30,40 @@ def self.stop_service(id)
     sudo 'service', id.to_s, 'stop' if OS.linux? && CDO.chef_managed
   end
 
+  # We've been having problems with 'sudo service dashboard stop', where it
+  # gets hung waiting for the process to stop, but the signal never takes effect.
+  # This calls and retries stop-with-status, which will wait for a bit but
+  # return an error code if the service doesn't actually stop. It finishes with
+  # a call to the normal stop method, which will wait indefinitely, if the retries
+  # all fail - this allows us to manually go in and kill the process without needing
+  # to restart the build.
+  def self.stop_service_with_retry(id, retry_count)
+    if OS.linux? && CDO.chef_managed
```

Comment:
Why do we need the `OS.linux?` check? Do we expect non-linux, non-developer environments? Just curious.